fdb-extensions/src/main/java/com/apple/foundationdb/async/hnsw/AccessInfo.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * AccessInfo.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.async.hnsw;
+
+import com.apple.foundationdb.linear.RealVector;
+import com.google.common.base.Objects;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Class to capture the current state of this HNSW that cannot be expressed as metadata but that also is not the actual
+ * data that is inserted, organized and retrieved. For instance, an HNSW needs to keep track of the entry point that
+ * resides in the highest layer(currently). Another example is any information that pertains to coordinate system
+ * transformations that have to be carried out prior/posterior to inserting/retrieving an item into/from the HNSW.
+ */
+class AccessInfo {
+    /**
+     * The current entry point. All searches start here.
+     */
+    @Nonnull
+    private final EntryNodeReference entryNodeReference;
+
+    /**
+     * A seed that can be used to reconstruct a random rotator {@link com.apple.foundationdb.linear.FhtKacRotator} used
+     * in ({@link StorageTransform}.
+     */
+    private final long rotatorSeed;
+
+    /**
+     * A centroid that is usually derived as an average over some vectors seen so far. It is used to create the
+     * {@link StorageTransform}.
+     */
+    @Nullable
+    private final RealVector centroid;
```

Comment:
One thing that confused me is that this is actually the _rotated anti-centroid_, not the centroid. That is to say, once we rotate the user vectors, we can add this value to center them over the origin. I see in HNSW that we are, in fact, correctly computing it (multiplying the centroid by negative one and then rotating), but it would be helpful if we noted that in the name of the variable here. We could also go with `translationVector` or something, though the comments should probably note that this should be the rotated anti-centroid